### ISSUE TITLE
Building error with latest pcl-trunk: emmintrin.h (SSE2) is included for armv7l platform.

### ISSUE THREAD

**Issue by lrvk123** [PointCloudLibrary/pcl#1271]
Building pcl-trunk fails with
/home/ubuntu/Build/pcl-trunk/gpu/people/src/bodyparts_detector.cpp:46:23: fatal error: emmintrin.h: No such file or directory
 #include "emmintrin.h"
                       ^
compilation terminated.
make[2]: **\* [gpu/people/CMakeFiles/pcl_gpu_people.dir/src/bodyparts_detector.cpp.o] Error 1
make[2]: Leaving directory `/home/ubuntu/Build/pcl-trunk/build'
make[1]: *** [gpu/people/CMakeFiles/pcl_gpu_people.dir/all] Error 2
make[1]: Leaving directory`/home/ubuntu/Build/pcl-trunk/build'
make: **\* [all] Error 2

**Comment by lrvk123** [comment]
I found that it was my mistake: PCL_ENABLE_SSE was "ON".

**Comment by taketwo** [comment]
Strange, because CMake is supposed to check for SSE support before adding corresponding compiler flags (see [here](https://github.com/PointCloudLibrary/pcl/blob/master/cmake/pcl_find_sse.cmake#L117)).

**Comment by sjdrc** [comment]
I'm attempting to compile PCL 1.8 on the Jetson TX1 and am running into the same issue, even after ensuring that PCL_ENABLE_SSE is set to "OFF". My application requires the gpu_people module on the ARM platform, does anyone have any ideas on how I can make this work?

**Comment by sjdrc** [comment]
After adding an empty ```emmintrin.h``` file in the build/include folder the code compiles correctly, but I haven't yet tested to see if gpu_people works.

**Comment by rghamilton3** [comment]
I hate to resurrect this but I'm having the same issue on my TX1 running the latest Ubuntu 16.04 installed using JetPack 3.0. I've set the PCL_ENABLE_SSE flag to off. I've tried deleting everything in the build folder a few times and still no luck.

**Comment by SergioRAgostinho** [comment]
tl;dr: gpu_people does not support ARM. Have a look at this https://github.com/OpenPTrack/open_ptrack  from one of the guys who wrote the people module on PCL. This one even support compilation on a Jetson Tx1

> I found that it was my mistake: PCL_ENABLE_SSE was "ON". However, when building on ARM platform the option is only shown in 'advanced' view and it is switched on by default.

This just makes CMake test if your cpu supports SSE. The only flag that get's set after that it is supposedly 
```cmake
# Setting -ffloat-store to alleviate 32bit vs 64bit discrepancies on non-SSE
# platforms.
set(SSE_FLAGS "-ffloat-store")
```

Here's the thing: gpu_people apparently depends on having SSE2 capabilities, because that source file explicitly includes a header (emmintrin.h) which needs to be compiled with that instruction set. ARM does not support SSEx.
